lib/level/connections.ex

```diff
@@ -102,6 +103,34 @@ defmodule Level.Connections do
     GroupMemberships.get(user, struct(GroupMemberships, args), info)
   end
 
+  @doc """
+  Fetches the current user's membership.
+  """
+  @spec group_membership(Group.t(), map(), authenticated_context()) ::
+          {:ok, GroupUser.t()} | {:error, String.t()}
+  def group_membership(%Group{} = group, _args, %{context: %{current_user: user}}) do
+    case Spaces.get_space(user, group.space_id) do
+      {:ok, %{space_user: space_user, space: space}} ->
+        case Level.Groups.get_group_membership(group, space_user) do
+          {:ok, group_user} ->
+            {:ok, group_user}
+
+          _ ->
+            virtual_group_user = %GroupUser{
+              state: "NOT_SUBSCRIBED",
+              space: space,
+              group: group,
+              space_user: space_user
+            }
+
+            {:ok, virtual_group_user}
+        end
+
+      error ->
+        error
```

Comment:
@devonestes I think this should be clearer now. If there's an error fetching the space, that will propagate as an error tuple. If the group membership exists, return it; otherwise, return a virtual record with hard-coded state (instead of `nil`).

Reply:
yeah, that makes much more sense to me! And it will probably avoid a bunch of nil checks later on, which is nice. Although that's not as much of an issue in Elixir than it is in Ruby thanks to easy pattern matching and dialyzer 😄